grouper/fe/util.py

```diff
@@ -157,17 +164,31 @@ def get_current_user(self):
 
         # service accounts are, by definition, not interactive users
         if user.is_service_account:
-            raise InvalidUser()
+            raise InvalidUser("{} is a service account".format(username))
 
         return user
 
     def prepare(self):
         # type: () -> None
-        if not self.current_user or not self.current_user.enabled:
+        username = self.request.headers.get(settings().user_auth_header)
+        try:
+            user = self.get_or_create_user(username)
+        except InvalidUser as e:
+            self.set_status(403)
+            self.raise_and_log_exception(tornado.web.HTTPError(403))
+            self.render(
+                "errors/baduser.html", message=str(e), how_to_get_help=settings().how_to_get_help
+            )
+            self.finish()
+            return
+
+        if not user or not user.enabled:
```

Comment:
Maybe make this also go to the baduser page instead of forbidden

Reply:
Yeah, thinking about this some more, that's a good idea. Will add that in before landing.